pkg/apis/uiplugin/v1alpha1/types.go

```diff
@@ -31,20 +31,73 @@ type UIPluginList struct {
 	Items           []UIPlugin `json:"items"`
 }
 
-// +kubebuilder:validation:Enum=Dashboards
+// +kubebuilder:validation:Enum=Dashboards;TroubleshootingPanel;DistributedTracing
 type UIPluginType string
 
 const (
 	// TypeDashboards deploys the Dashboards Dynamic Plugin for OpenShift Console.
 	TypeDashboards UIPluginType = "Dashboards"
+	// DistributedTracing deploys the Distributed Tracing Dynamic Plugin for the OpenShift Console
+	TypeDistributedTracing UIPluginType = "DistributedTracing"
+	// TroubleshootingPanel deploys the Troubleshooting Panel Dynamic Plugin for the OpenShift Console
+	TypeTroubleshootingPanel UIPluginType = "TroubleshootingPanel"
 )
 
+// TroubleshootingPanelConfig contains options for configuring the Troubleshooting Panel  plugin
+type TroubleshootingPanelConfig struct {
+	// Timeout is the max duration before a query timeout
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OCP Console Query Timeout",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:ocpConsoleTimeout"}
+	// +kubebuilder:validation:Pattern:="^([0-9]+)([smhd]{0,1})$"
+	Timeout string `json:"timeout,omitempty"`
+	// korrel8r defines the Korrel8r instance that the troubleshooting panel plugin will connect to
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Korrel8r Instance"
+	Korrel8r TroubleshootingPanelKorrel8rConfig `json:"korrel8r,omitempty"`
+}
+
+type TroubleshootingPanelKorrel8rConfig struct {
+	// Name of the korrel8r instance
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Korrel8r Instance Name"
+	Name string `json:"name,omitempty"`
+
+	// Namespace of the korrel8r instance
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Korrel8r Instance Namespace"
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// DistributedTracingConfig contains options for configuring the Distributed Tracing plugin
+type DistributedTracingConfig struct {
+	// Timeout is the max duration before a query timeout
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OCP Console Query Timeout",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:ocpConsoleTimeout"}
+	// +kubebuilder:validation:Pattern:="^([0-9]+)([smhd]{0,1})$"
```

Comment:
I am not very good in regular expressions, but this allows me to create e.g:
```
spec:
  distributedTracing:
    timeout: '5'
```
so perhaps `^([0-9]+)([smhd]{1})$` ?